javaslang/src/main/java/javaslang/collection/PriorityQueue.java

```diff
@@ -46,7 +48,20 @@ private PriorityQueue(SerializableComparator<? super T> comparator, List<Node<Ra
      */
     @Override
     public PriorityQueue<T> enqueue(T element) {
-        return with(insertForest(comparator, forest, element));
+        final List<Node<Ranked<T>>> insert = insert(comparator, forest, element);
+        return with(insert, size + 1);
+    }
+
+    /**
+     * Enqueues the given elements.
+     *
+     * @param elements An {@link PriorityQueue} of elements, may be empty
+     * @return a new {@link PriorityQueue} instance, containing the new elements
+     * @throws NullPointerException if elements is null
+     */
+    @Override
+    public PriorityQueue<T> enqueueAll(Iterable<? extends T> elements) {
+        return merge(ofAll(comparator, elements));
```

Comment:
if we make a small queue of the elements and merge it with the original (instead of adding them one-by-one to the original), the complexity will depend on the number of added elements and not on the size of the original tree (I think :)) ... it got quicker, so I guess I'm right :p)

Now the tests show that it's only `4.4x` slower than the reference impl :D